quickannotator/api/v1/annotation/__init__.py

```diff
@@ -185,6 +212,27 @@ def get(self, args, image_id, annotation_class_id, tile_id):
         """
         result = get_annotations_for_tile(image_id, annotation_class_id, tile_id, args['is_gt'])
         return result, 200
+    
+@bp.route('/<int:image_id>/<int:annotation_class_id>/withinpoly')
+class AnnotationsWithinPolygon(MethodView):
+    @bp.arguments(GetAnnWithinPolyArgsSchema, location='json')
+    @bp.response(200, AnnRespSchema(many=True))
+    def post(self, args, image_id, annotation_class_id):
+        """     get all annotations within a polygon
+        """
+
+        # 1. Get all tiles intersecting the polygon
+        tiles_ids_within_mask, _, _ = get_tile_ids_intersecting_mask(image_id, annotation_class_id, constants.MASK_DILATION)
```

Comment:
wow, lot of queries here -- can you walk me through the logic? i would have expected to be able to e.g., send the polygon to the database and perform the intersection there in the GEOM space. i suspect with the spatial index that would be extremely quick - on par with using the tile_id to pre-filter?

Reply:
Yeahhh... this should definitely be done in the database. I wasn't sure if sqlite would actually use the spatial index automatically, as we previously observed. That said, if we add a tileid filter to the spatial query, we can still get reasonable performance with sqlite while still using a db agnostic query.

I'm going to fill in the `get_annotations_within_polygons()` function below to perform the spatial query directly on the database. Sound good?

Reply:
yup!